### PR TITLE
ICU-21055 Remove test inputs causing logKnownIssues test run to hang

### PR DESCRIPTION
**Pull request by echeran**
##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21055
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [ ] Documentation is changed or added

I filed ICU-21080 to capture the outstanding work needed to restore the input values into the test case.

### CODE REVIEW

**Comment by FrankYFTang** [comment]
loop in @sffc sffc since he has some opinion about this.

File: icu4c/source/test/intltest/restsnew.cpp [hunk 1]
```diff
@@ -1395,6 +1396,15 @@ void NewResourceBundleTest::TestFilter() {
...
 }
 
+/*
+ * The following test for ICU-20706 causes infinte loops on certain inputs for
+ * locales and calendars.  In order to unblock the build (ICU-21055), the test
+ * is comment out.
+ * The issue of the infinite loops and its blocking dependencies were captured
+ * in ICU-21080.
+ */
+#if 0
```
Comment (markusicu) [inline icu4c/source/test/intltest/restsnew.cpp:1406]: Please limit the `#if 0 ... #endif` to just the two blocks of test values that "will cause infinity loop". We want to keep testing the ones that do terminate.

File: icu4c/source/test/intltest/restsnew.cpp [hunk 2]
```diff
@@ -1395,13 +1396,22 @@ void NewResourceBundleTest::TestFilter() {
     }
 }
 
+/*
+ * The following test for ICU-20706 has infinte loops on certain inputs for
```
Comment (markusicu) [inline icu4c/source/test/intltest/restsnew.cpp:1400]: infinte --> infinite

File: icu4c/source/test/intltest/restsnew.cpp [hunk 3]
```diff
@@ -1395,13 +1396,22 @@ void NewResourceBundleTest::TestFilter() {
...
+ */
+
 void NewResourceBundleTest::TestIntervalAliasFallbacks() {
     const char* locales[] = {
         // Thee will not cause infinity loop
         "en",
         "ja",
 
         // These will cause infinity loop
+        #if 0
```
Comment (markusicu) [inline icu4c/source/test/intltest/restsnew.cpp:1414]: Please remove the indents from the four lines with `#`.

File: icu4c/source/test/intltest/restsnew.cpp (hunk 1 above)
Comment (echeran) [reply icu4c/source/test/intltest/restsnew.cpp]: Oh, right, of course.  Done.

File: icu4c/source/test/intltest/restsnew.cpp (hunk 2 above)
Comment (echeran) [reply icu4c/source/test/intltest/restsnew.cpp]: Done.

File: icu4c/source/test/intltest/restsnew.cpp (hunk 3 above)
Comment (echeran) [reply icu4c/source/test/intltest/restsnew.cpp]: Done.

**Comment by sffc** [comment]
LGTM, though I must say that I don't fully understand the desire to make `-K` run to completion.

**Pull request merged**